linux/keyman-config/tests/canonical_language_code_utils_tests.py

```diff
@@ -85,3 +85,56 @@ def test_FindBestTag_NotAddingRegion(self):
                         # Verify
                         msg = "\nFor %s:\nExpected: %s\ngot:      %s " % (tag, expectedTag, bestTag)
                         self.assertEqual(expectedTag, bestTag, msg)
+
+
+class NormalizeLanguageTests(unittest.TestCase):
+    def test_normalizeLanguage(self):
+        # Setup
+        languages = [
+            {'id': 'de'},
+            {'id': 'esi-Latn'},
+            {'id': 'dyo'},
+            {'id': 'fuh-Arab'}
+        ]
+
+        for testcase in [
+            {'given': 'de', 'expected': 'de'},
+            {'given': 'esi', 'expected': 'esi'},
+            {'given': 'esi-Latn', 'expected': 'esi'},
+            {'given': 'es', 'expected': None},
+            {'given': 'en', 'expected': None},
+            {'given': None, 'expected': 'de'},
```

Comment:
Why is `de` expected?

Reply:
`'de'` is the first language in the list, so should be the default if `None` is given